fs/fs.c

```diff
@@ -518,8 +518,7 @@ static int fs_read_lmb_check(const char *filename, ulong addr, loff_t offset,
 	if (lmb_alloc_addr(&lmb, addr, read_len) == addr)
 		return 0;
 
-	log_err("** Reading file would overwrite reserved memory **\n");
-	return -ENOSPC;
+	return 0;
```

Comment:
CONFIG_LMB_USE_MAX_REGIONS fixes the build issue but LMB is still checking reserved memory.


> This call succeeds when the return is addr which means it got non reserved memory. 
> If it in reserved memory it returns 0 which means the allocation failed. 
> The removed lines of code still seem correct. What is the rationale for returning 0? Where is the dependency?
> 

Yes, the removed lines are correct. But in our case, we want to load some files out of on chip memory (CONFIG_SYS_SDRAM_BASE). 

Plus, **lmb_alloc_addr** returns an address on where we should allocate for the request memory. But the allocated memory does not use for any other reason rather than checking the reserved memory. We can bypass this checking by **return 0** and the functionality still doesn't change.

Check the below function for more details: [fs/fs.c#L535](https://github.com/Renesas-SST/u-boot/blob/dunfell/rz-sbc/fs/fs.c#L535)
The codeflow: _fs_read -> fs_read_lmb_check -> lmb_alloc_addr
```
static int _fs_read(const char *filename, ulong addr, loff_t offset, loff_t len,
		    int do_lmb_check, loff_t *actread)
{
	struct fstype_info *info = fs_get_info(fs_type);
	void *buf;
	int ret;

#ifdef CONFIG_LMB
	if (do_lmb_check) {
		ret = fs_read_lmb_check(filename, addr, offset, len, info);
		if (ret)
			return ret;
	}
#endif

	/*
	 * We don't actually know how many bytes are being read, since len==0
	 * means read the whole file.
	 */
	buf = map_sysmem(addr, len);
	ret = info->read(filename, buf, offset, len, actread);
	unmap_sysmem(buf);

	/* If we requested a specific number of bytes, check we got it */
	if (ret == 0 && len && *actread != len)
		log_debug("** %s shorter than offset + len **\n", filename);
	fs_close();

	return ret;
}
```

> If you are always returning 0, there is no point in doing CONFIG_LMB. just disable the macro instead of returning false positives.


Yes, I think I'm better remove checking section in _fs_read than returning false positive.
```
#if 0
	if (do_lmb_check) {
		ret = fs_read_lmb_check(filename, addr, offset, len, info);
		if (ret)
			return ret;
	}
#endif
```